### PR TITLE
#7532 Adding model save support for Sentiment Analysis model

### PR DESCRIPTION
**Pull request by Naveenkhasyap**
Added model save option once we train the model on IMDB Data set 
1. Save model checkpoint with best accuracy while training (epoch based)
2. Save the whole model (big file)

Let me know if any changes needed further I have trained and tested the same and used the model for inferencing as well 
while using the saved model for prediction create CNN with same voc_size as you have set while training the model.

### CODE REVIEW

**Comment by Naveenkhasyap** [comment]
@googlebot I signed it!

File: research/sentiment_analysis/sentiment_main.py [hunk 1]
```diff
@@ -48,9 +50,23 @@ def run_model(dataset_name, emb_dim, voc_size, sen_len,
...
+
+
   model.fit(x_train, y_train, batch_size=batch_size,
-            validation_split=0.4, epochs=epochs)
+            validation_split=0.4, epochs=epochs,callbacks=[checkpoint_callback])
+
   score = model.evaluate(x_test, y_test, batch_size=batch_size)
+  
+  model.save(model_save_dir+"/full-model.h5")
+
```
Comment (saberkun) [inline research/sentiment_analysis/sentiment_main.py]: use os.path.join?

File: research/sentiment_analysis/sentiment_main.py [hunk 2]
```diff
@@ -85,8 +101,11 @@ def run_model(dataset_name, emb_dim, voc_size, sen_len,
                       help="The number of epochs for training.",
                       type=int, default=55)
 
+  parser.add_argument("-f", "--folder",
+                      help="folder/dir to save trained model",
+                      type=str, default=os.getcwd())
```
Comment (saberkun) [inline research/sentiment_analysis/sentiment_main.py]: default None and raise error?

File: research/sentiment_analysis/sentiment_main.py [hunk 3]
```diff
@@ -48,9 +50,23 @@ def run_model(dataset_name, emb_dim, voc_size, sen_len,
...
+  
+  checkpoint_callback = tf.keras.callbacks.ModelCheckpoint(filepath, monitor='val_accuracy',
+                                            verbose=1,save_best_only=True, 
+                                            save_weights_only=True,mode='auto')
+
+
   model.fit(x_train, y_train, batch_size=batch_size,
-            validation_split=0.4, epochs=epochs)
+            validation_split=0.4, epochs=epochs,callbacks=[checkpoint_callback])
+
```
Comment (saberkun) [inline research/sentiment_analysis/sentiment_main.py]: add a space before

File: research/sentiment_analysis/sentiment_main.py [hunk 4]
```diff
@@ -15,12 +15,14 @@
...
 import sentiment_model
+import os
+
 
 _DROPOUT_RATE = 0.95
 
 
 def run_model(dataset_name, emb_dim, voc_size, sen_len,
-              hid_dim, batch_size, epochs):
+              hid_dim, batch_size, epochs,model_save_dir):
```
Comment (saberkun) [inline research/sentiment_analysis/sentiment_main.py]: add a space before

File: research/sentiment_analysis/sentiment_main.py [hunk 5]
```diff
@@ -15,12 +15,14 @@
 
 from data import dataset
```
Comment (saberkun) [inline research/sentiment_analysis/sentiment_main.py]: please sort import

File: research/sentiment_analysis/sentiment_main.py [hunk 6]
```diff
@@ -85,8 +101,11 @@ def run_model(dataset_name, emb_dim, voc_size, sen_len,
...
 
+  parser.add_argument("-f", "--folder",
+                      help="folder/dir to save trained model",
+                      type=str, default=os.getcwd())
   args = parser.parse_args()
 
   run_model(args.dataset, args.embedding_dim, args.vocabulary_size,
             args.sentence_length, args.hidden_dim,
-            args.batch_size, args.epochs)
+            args.batch_size, args.epochs,args.folder)
```
Comment (saberkun) [inline research/sentiment_analysis/sentiment_main.py]: add a space

**Comment by jaeyounkim** [comment]
@Naveenkhasyap Please make suggested changes.

File: research/sentiment_analysis/sentiment_main.py (hunk 1 above)
Comment (Naveenkhasyap) [reply research/sentiment_analysis/sentiment_main.py:70]: Done

File: research/sentiment_analysis/sentiment_main.py (hunk 2 above)
Comment (Naveenkhasyap) [reply research/sentiment_analysis/sentiment_main.py]: updated, instead of raising error i called parser.error so it will take care of raising error with message passed to it.

File: research/sentiment_analysis/sentiment_main.py (hunk 3 above)
Comment (Naveenkhasyap) [reply research/sentiment_analysis/sentiment_main.py:66]: ✅

File: research/sentiment_analysis/sentiment_main.py (hunk 4 above)
Comment (Naveenkhasyap) [reply research/sentiment_analysis/sentiment_main.py]: ✅

File: research/sentiment_analysis/sentiment_main.py (hunk 5 above)
Comment (Naveenkhasyap) [reply research/sentiment_analysis/sentiment_main.py:17]: ✅ replaced it along with system import argparse

File: research/sentiment_analysis/sentiment_main.py (hunk 6 above)
Comment (Naveenkhasyap) [reply research/sentiment_analysis/sentiment_main.py]: ✅

**Comment by Naveenkhasyap** [comment]
@saberkun updated the code with review changes , Have a look.

**Comment by saberkun** [comment]
Thanks! Merged.
